pkg/k8sutil/nodes.go

```diff
@@ -0,0 +1,22 @@
+package k8sutil
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GetAnyNodeLabels returns the labels of an arbitrary node. This is useful
+// to extract global labels such as the cluster name.
+func GetAnyNodeLabels(ctx context.Context, client kubernetes.Interface) (map[string]string, error) {
```

Comment:
This seems rather specific to the use-case we have the azure metadata stuff, not sure it should be within `k8sutil` but rather be a helper within metadata file.

Reply:
The same method can be used for EKS (but only when created via eksctl). But we can move it to metadata.